components/store_state.py

```diff
@@ -26,47 +28,169 @@ def load_state_from_store(store_data, key):
             return store_data[key]
         return None
 
+    @staticmethod
+    def parse_url(url):
+        """Parse URL and extract useful information."""
+        url = str(url)
+        parsed_url = urlparse(url)
+        query_params = parse_qs(parsed_url.query)
+        path_parts = parsed_url.path.strip("/").split("/")
+
+        return {
+            "path": parsed_url.path,
+            "query_params": query_params,
+            "fragment": parsed_url.fragment,
+            "path_parts": path_parts
+        }
+
     def setup_dash_callbacks(self, app, store_id):
         """Set up Dash callbacks to manage dcc.Store data, supporting multiple dynamic input components."""
 
         @app.callback(
             Output(store_id, "data"),
-            Input({"type": "dynamic-input", "index": ALL}, "value"),
+            [Input(id, "value") for id in self.persist_ids],
+            [Input("url", "href")],
             State(store_id, "data"),
             prevent_initial_call=True,
         )
-        def save_data(input_values, store_data):
-            # Update storage if triggered by input component
+        def manage_store_data(*args):
+            # The URL is the second-to-last argument
+            url = args[-2]
```

Comment:
I do not like to use positional arguments, what -2 refers to? can we avoid doing this?